bugsnag-android-core/src/main/java/com/bugsnag/android/internal/ImmutableConfig.kt

```diff
@@ -181,6 +182,34 @@ internal fun convertToImmutableConfig(
     )
 }
 
+private fun validateApiKey(value: String) {
+    if (isInvalidApiKey(value)) {
+        w(
```

Comment:
This should log to the configured `Logger` rather than always directing to the `DebugLogger`

```suggestion
        logger.w(
```